### DISCUSSION TITLE
Has emotion support only existed for coqui API users?

### DISCUSSION THREAD

**Discussion by matbeedotcom** [idiap/coqui-ai-TTS#41] [Ideas]
What would it take to get emotion control out of this?

**Comment by eginhard** [comment]
Yes, the `emotion` arguments in https://github.com/idiap/coqui-ai-TTS/blob/a682fa8d56cb3eba131fcb45e0623c38132e63a0/TTS/api.py#L243 could only be used with the proprietary Coqui Studio models, so they are no longer useful. If someone adds a model that supports generating emotional speech, they could be hooked up to that.

**Comment by IIEleven11** [comment]
Theoretically XTTSv2 should be able to do this because it is a child of the tortoise model and this is where that emotion control comes from. I've been trying to decipher this over the last few days and I found my way here to see if anyone else has any information. Along with some other things it appears intentionally obfuscated/omitted. The tokenizer/vocab show little to help. I am maybe jumping to conclusions though, if anyone has any insight please let me know.

**Comment by agonzalezd** [comment]
An option could be adding another embedder for tags, similar to a speaker embedding, only for emotions. This embedder would simply generate an embedding for a label instead of using another neural model for computing this feature, using [this](https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html), for example. A small embedding could do it, since no more than 10 emotions will be used in a dataset (hopefully). The embedding dimesion could be updated in the model configuration.
